File: computation numeric/secant/Python/secant.py
import math

def e(x):
    fx = math.exp(-x) - x
    return fx

def secant(x1, x2, max_step = 20, tol = 0.0001):
    
    loopCount = 1
    
    for loopCount in range(max_step):
        
        p = x2 - (e(x2) * ((x1 - x2) / (e(x1) - e(x2))))
        
        print('f(x',loopCount - 1,')', e(x1), 'f(x',loopCount,')', e(x2))
        
        print('x',int(loopCount + 1), '=', p, '\n')
        
        if (abs(e(p)) < tol):
            print('root is', p, 'iteration', loopCount + 1)
            return
        
        x1 = x2
        x2 = p
    
    print('Root find stopped at', p)
        
    
secant(0, 1)

File: computation numeric/secant/PHP/secant.php
<?php 
// PHP Program to find root of an 
// equations using secant method 

// function takes value of x 
// and returns f(x) 
function f( $x) 
{ 
	$f = exp(-$x) - $x; 
	return $f; 
} 

function secant($x1, $x2, $E) 
{ 
	$n = 0; $xm; 
	$x0; $c; 
	if (f($x1) * f($x2) < 0) 
	{ 
		do { 
			
			// calculate the intermediate value 
			$x0 = ($x1 * f($x2) - $x2 * 
				f($x1)) / (f($x2) - f($x1)); 

			// check if x0 is root 
			// of equation or not 
			$c = f($x1) * f($x0); 

			// update the value of interval 
			$x1 = $x2; 
			$x2 = $x0; 

			// update number of iteration 
			$n++; 

			// if x0 is the root of equation 
			// then break the loop 
			if ($c == 0) 
				break; 
			$xm = ($x1 * f($x2) - $x2 * f($x1)) / 
							(f($x2) - f($x1)); 
								
		// repeat the loop 
		// until the convergence 
		} while (abs($xm - $x0) >= $E); 

		echo "Root of the given equation=". $x0."\n" ; 
		echo "No. of iterations = ". $n ; 
		
	} else
		echo "Can not find a root in the given inteval"; 
} 

// Driver code 
{ 
	
	// initializing the values 
	$x1 = 0; $x2 = 1; 
	$E = 0.0001; 
	secant($x1, $x2, $E); 
	return 0; 
}  
?> 
